docs/src/assets/scss/components/theme-switcher.scss

```diff
@@ -20,7 +20,8 @@
 }
 
 .theme-switcher__button {
-    flex: 0;
+    flex-shrink: 1;
+    flex-grow: 1;
```

Comment:
Did you try in all device width? I see that this fails for an intermediate device width?

Reply:
Also I'd rather suggest so that the button wraps the text to the next line instead of pushing the second character sometimes
```suggestion
    flex-wrap: wrap;
```

Reply:
> Did you try in all device width? I see that this fails for an intermediate device width?

The main point of this pull request is to fix the issue shown in the screenshot I had provided, i.e., when there're enough spaces buttons should take up them equally.

Regarding how to deal with it when there're not enough space, `flex-shrink` is the original implement, `flex-wrap` is another good option - if we all agree on this one, I can update it.

Reply:
> The main point of this pull request is to fix the issue shown in the screenshot I had provided, i.e., when there're enough spaces buttons should take up them equally.

Yea, But still we aren't supposed to wrap the text? Im not how it works in Chinese (may be if its two word then wrap is okayish else nope) but somewhere in device width around `814` I still see button being this way. Also seems like padding remains the same for large designs and smaller ones which forces the content to either wrap when its long. 
<img width="182" alt="Screenshot 2023-01-08 at 11 07 08 AM" src="https://user-images.githubusercontent.com/32865581/211182109-ea0ef6f5-e15a-47f5-b1a7-74697d9b7245.png">

Reply:
Not sure we're on the same page.

There're two cases here:

1. when there're enough spaces, flex items should grow equally - hopefully we all agree on this.
2. when there aren't enough spaces, to shrink, or to wrap. The original code chose to shrink, and I kept the behavior in this pull request, while you suggest applying `flex-wrap` to wrap, right?

But it seems you thought there were third case here?

Reply:
> Im not how it works in Chinese (may be if its two word then wrap is okayish else nope) but somewhere in device width around `814` I still see button being this way.

They are one word, I think it should wrap, using `flex-wrap: wrap` would be better.